src/VisualStudio/Core/Def/Implementation/InheritanceMargin/MarginGlyph/InheritanceMargin.xaml

```diff
@@ -215,7 +84,172 @@
     </Button.Template>
 
     <Button.ContextMenu>
-        <!-- Ensure the context menu is not null -->
-        <ContextMenu Style="{StaticResource ContextMenuStyle}"/>
+        <ContextMenu ItemsSource="{Binding MenuItemViewModels}"
+                     Style="{StaticResource ContextMenuStyle}"
+                     ItemContainerTemplateSelector="{StaticResource TemplateSelector}"
+                     UsesItemContainerTemplate="True">
+            <ContextMenu.Resources>
+                <!-- Style used to display a single target menu item -->
+                <Style x:Key="TargetMenuItemStyle" TargetType="{x:Type MenuItem}">
+                    <Setter Property="Icon" Value="{StaticResource NonSharedIcon}"/>
+                    <Setter Property="Header" Value="{Binding  DisplayContent}"/>
+                    <Setter Property="AutomationProperties.Name" Value="{Binding AutomationName}"/>
+                    <Setter Property="IsCheckable" Value="False"/>
+                    <EventSetter Event="Click" Handler="TargetMenuItem_OnClick"/>
+                    <Setter Property="HorizontalContentAlignment" Value="{Binding Path=HorizontalContentAlignment, RelativeSource={RelativeSource AncestorType={x:Type ItemsControl}}}"/>
+                    <Setter Property="VerticalContentAlignment" Value="{Binding Path=VerticalContentAlignment, RelativeSource={RelativeSource AncestorType={x:Type ItemsControl}}}"/>
+                    <Setter Property="Background" Value="Transparent"/>
+                        <!-- Style copied from editor -->
+                    <Setter Property="Foreground" Value="{DynamicResource {x:Static vsui:EnvironmentColors.CommandBarTextActiveBrushKey}}"/>
+                    <Setter Property="DockPanel.Dock" Value="Top"/>
+                    <Setter Property="Padding" Value="0,2,0,2"/>
+                    <Setter Property="Template">
+                        <Setter.Value>
+                            <ControlTemplate TargetType="{x:Type MenuItem}">
+                                <Grid SnapsToDevicePixels="true" Background="Transparent" MinHeight="22">
+                                    <Grid.ColumnDefinitions>
+                                        <ColumnDefinition Width="Auto" SharedSizeGroup="MenuItemIconColumnGroup"/>
+                                        <ColumnDefinition Width="*"/>
+                                        <ColumnDefinition Width="Auto" SharedSizeGroup="MenuItemIGTColumnGroup"/>
+                                        <ColumnDefinition Width="23" SharedSizeGroup="MenuItemExpanderColumnGroup"/>
+                                    </Grid.ColumnDefinitions>
+
+                                    <Rectangle Name="Bg" Stroke="Transparent" Fill="Transparent" StrokeThickness="0" Grid.ColumnSpan="5"/>
+                                    <ContentPresenter Grid.Column="0" x:Name="Icon" Margin="{Binding Margin}" Width="16" Height="16" VerticalAlignment="Center"
```

Comment:
The margin here is a biding, previously it is 4, 1, 4, 1,
It is changed to achieve the indentation effect.
![Uploading image.png…]()

Reply:
The value is binding to the Margin property in TargetViewModel